lambdas/evaluate-gpg45-scores/src/main/java/uk/gov/di/ipv/core/evaluategpg45scores/EvaluateGpg45ScoresHandler.java

```diff
@@ -174,21 +173,26 @@ private Map<String, Object> buildJourneyErrorResponse(ErrorResponse errorRespons
     }
 
     @Tracing
-    private boolean hasMatchingGpg45Profile(
+    public boolean hasMatchingGpg45Profile(
             List<VerifiableCredential> vcs,
             IpvSessionItem ipvSessionItem,
             ClientOAuthSessionItem clientOAuthSessionItem,
             String ipAddress,
             String deviceInformation)
             throws UnknownEvidenceTypeException, SqsException, CredentialParseException {
         if (!userIdentityService.checkRequiresAdditionalEvidence(vcs)) {
-            Gpg45Scores gpg45Scores = gpg45ProfileEvaluator.buildScore(vcs);
-            Optional<Gpg45Profile> matchedProfile =
+            var gpg45Scores = gpg45ProfileEvaluator.buildScore(vcs);
+
+            var requestedVotsByStrength = clientOAuthSessionItem.getRequestedVotsByStrength();
+            var supportedGpg45ProfilesByVotStrength =
+                    requestedVotsByStrength.stream()
+                            .flatMap(vot -> vot.getSupportedGpg45Profiles().stream())
+                            .toList();
+            var matchedProfile =
                     gpg45ProfileEvaluator.getFirstMatchingProfile(
-                            gpg45Scores, Vot.P2.getSupportedGpg45Profiles());
+                            gpg45Scores, supportedGpg45ProfilesByVotStrength);
```

Comment:
If the `vtr` is ["P1", "P2"] does this match in descending strength, so if a P2 profile is matched it will be returned first?

Reply:
Yes so `getRequestedVotsByStrength` returns (supported) Vots in order of descending strength. Please see the test `shouldReturnJourneyMetForMeetingMediumAndLowConfidences` for that example. It references `P1_and_P2_profiles` which is `List.of(M1A, M1B, M2B, L1A)` (strength-ordered)